fix(components): [tree] filter update scheduling error

a92126 opened the pull request:
fix(components): [tree] fix bug #20259 
Ensure the sequence for the traverse step. If call the traverse without await, the code below will run before the children had been processed, making the parent node invisible.

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

a92126 committed:
Update tree-store.ts

fix: the bug #20259 
Ensure the sequence for the traverse step.

a92126 committed:
Merge branch 'dev' into patch-1

Dsaquel commented:
Could you add a `await sleep()` [here](https://github.com/a92126/element-plus/blob/b6aa77a4811b307f43818ae0f8d30b6741845d68/packages/components/tree/__tests__/tree.test.ts#L352) to fix the test ?

a92126 committed:
Merge branch 'dev' into patch-1

a92126 committed:
fix(components): [tree] filter update scheduling error

fix(components): [tree] fix bug #20259
Ensure the sequence for the traverse step. If call the traverse without await, the code below will run before the children had been processed, making the parent node invisible.

a92126 commented:
> Could you add a `await sleep()` [here](https://github.com/a92126/element-plus/blob/b6aa77a4811b307f43818ae0f8d30b6741845d68/packages/components/tree/__tests__/tree.test.ts#L352) to fix the test ?

Thanks, I change the test case to sleep.

a92126 committed:
Merge branch 'dev' into patch-1

warmthsea reviewed (approved):
👍  Thanks, I just noticed this pr.
